solutions/silver/cses-1619.mdx

```diff
@@ -244,6 +244,37 @@ public class Customers {
 ```
 
 </JavaSection>
+<PySection>
+<Warning>
+
+Due to Python's big constant factor, this code will TLE on one test case.
+
+</Warning>
+
+```py
+n = int(input())
+
+times = []
+
+for _ in range(n):
+	start, end = map(int, input().split())
+	times.append((start, 1))
+	times.append((end, -1))
+
+times.sort()
+
+curr_ppl = 0
+max_ppl = 0
+
+for t in times:
+	curr_ppl += t[1]
+	max_ppl = max(max_ppl, curr_ppl)
+
+print(max_ppl)
+```
+
+</PySection>
```

Comment:
```suggestion
Since "all arrival and leaving times are distinct," we can store if a given time is an arrival or departure, and sort all the arrival and leaving times in one array as in the below solution:

```py
tmap = dict()  
times = []
for _ in range(int(input())):
	start, end = input().split()
	start = int(start)
	end = int(end)
	tmap[start] = 1
	tmap[end] = -1
	times.append(start)
	times.append(end)

times.sort()

curr_ppl = 0
max_ppl = 0
for t in times:
	curr_ppl += tmap[t]
	max_ppl = max(max_ppl, curr_ppl)
print(max_ppl)
```
</PySection>
```